### PR TITLE
feat: support zig `0.13.0` and setting default TTY at build time

### PR DESCRIPTION
**Pull request by jinzhongjia**
Feature: 
- Add zig `0.13.0` adaptation

It has not yet been tested whether it can be compiled with 12

### CODE REVIEW

**Comment by jinzhongjia** [comment]
What do you think of trying to add github action to support CI?
@AnErrupTion

**Review by AnErrupTion — APPROVED** [review]

**Comment by jinzhongjia** [comment]
Good CI can help ensure certain build compatibility
But if you only maintain compatibility with the latest release zig, there is no need.

**Comment by jinzhongjia** [comment]
After all, sometimes the person who initiated the PR may not do a complete version compatibility test.

**Comment by jinzhongjia** [comment]
The current problem is caused by clap in latest version being incompatible with version 12.
Need to open pr to clap

**Comment by AnErrupTion** [comment]
> The current problem is caused by clap in latest version being incompatible with version 12. Need to open pr to clap

I don't think clap maintains backwards compatibility with older versions of Zig either, you'll have to check with them on that.

**Comment by jinzhongjia** [comment]
acorrding this issue,https://github.com/Hejsil/zig-clap/issues/128

It seems that not to be backward compatible

**Comment by jinzhongjia** [comment]
ok , clap has agreed to maintain backward compatibility with current 12

**Comment by jinzhongjia** [comment]
We just need to wait for clap to merge pr.

**Comment by jinzhongjia** [comment]
Now clap's pr has been merged, `zigini` still has a little problem, but I have opened a PR

**Comment by jinzhongjia** [comment]
https://github.com/Kawaii-Ash/zigini/pull/1

**Comment by jinzhongjia** [comment]
We currently only need to wait for the PR merge to be completed. My local test of `0.12` and `0.13` has passed.

**Comment by jinzhongjia** [comment]
this issue: https://github.com/fairyglade/ly/issues/225

Added option for build: `default_tty`
@AnErrupTion please review my commit again. I'm not sure if it's correct.

File: build.zig [hunk 1]
```diff
@@ -15,13 +15,16 @@ comptime {
 const ly_version = std.SemanticVersion{ .major = 1, .minor = 1, .patch = 0 };
 var dest_directory: []const u8 = undefined;
 var data_directory: []const u8 = undefined;
+var default_tty: u8 = 0;
```
Comment (AnErrupTion) [inline build.zig:18]: You can make it `undefined` instead of `0`.

File: build.zig [hunk 2]
```diff
@@ -15,13 +15,16 @@ comptime {
...
 var exe_name: []const u8 = undefined;
 
 const ProgressNode = if (current_zig.minor == 12) *std.Progress.Node else std.Progress.Node;
 
 pub fn build(b: *std.Build) !void {
     dest_directory = b.option([]const u8, "dest_directory", "Specify a destination directory for installation") orelse "";
     data_directory = b.option([]const u8, "data_directory", "Specify a default data directory (default is /etc/ly). This path gets embedded into the binary") orelse "/etc/ly";
+    default_tty = b.option(u8, "default_tty", "set default tty") orelse 2;
+
```
Comment (AnErrupTion) [inline build.zig:27]: Minor nitpick: for consistency sake, please add proper capitalization (e.g. "Set default TTY").

**Review by AnErrupTion — CHANGES_REQUESTED** [review]

**Review by AnErrupTion — APPROVED** [review]

**Comment by jinzhongjia** [comment]
now, everything should be ok

**Comment by AnErrupTion** [comment]
Thanks for your contribution!

**Pull request merged**